tools/jdbc/src/main/java/org/duckdb/DuckDBResultSetMetaData.java

```diff
@@ -44,8 +44,9 @@ public DuckDBResultSetMetaData(int param_count, int column_count, String[] colum
     }
 
     public static DuckDBColumnType TypeNameToType(String type_name) {
-        if (type_name.endsWith("[]")) {
-            return DuckDBColumnType.LIST;
+        if (type_name.endsWith("]")) {
+            // VARCHAR[] or VARCHAR[2]
+            return type_name.endsWith("[]") ? DuckDBColumnType.LIST : DuckDBColumnType.ARRAY;
```

Comment:
We can also have arrays of lists or lists of arrays, e.g.:

```sql
select [[1], [2], [3]]::int[][3] as array_of_lists;
┌─────────────────┐
│ array_of_lists  │
│  integer[][3]   │
├─────────────────┤
│ [[1], [2], [3]] │
└─────────────────┘
select [[1, 2, 3], [4, 5, 6]]::int[3][] as list_of_arrays;
┌────────────────────────┐
│     list_of_arrays     │
│      integer[3][]      │
├────────────────────────┤
│ [[1, 2, 3], [4, 5, 6]] │
└────────────────────────┘

```

I wonder if this is handled correctly by this code (by unpacking recursively). Could we perhaps add a test?

Reply:
Ah nevermind, I see that it is tested in the test_all_types already